webserver/webserver/src/main/java/io/helidon/webserver/SocketConfiguration.java

```diff
@@ -259,6 +262,13 @@ default BackpressureStrategy backpressureStrategy() {
      */
     List<RequestedUriDiscoveryType> requestedUriDiscoveryTypes();
 
+    /**
+     * The allow list for trusted proxies.
+     *
+     * @return allow list for trusted proxies
+     */
+    AllowList trustedProxies();
```

Comment:
This must never return null. Either change to `Optional<AllowList>`, or return some default instance (by default this should be an allow list that denies all)

Reply:
If the application does not specify any trusted proxies (config or builder setting), the `build` method now creates a default one with a "deny" predicate that is always `true` which denies all.